src/Avalonia.Visuals/Rendering/DirtyVisuals.cs

```diff
@@ -89,19 +100,69 @@ public bool Remove(IVisual visual)
         /// <returns>A collection of visuals.</returns>
         public IEnumerator<IVisual> GetEnumerator()
         {
-            foreach (var i in _inner)
+            using (DeferChanges())
```

Comment:
Would it make more sense if ``GetEnumerator` returned the deferred changes immediately here after it had finished enumerating?

Reply:
It's a little harder. Wouldn't be susceptible to infinite loops? Also, how would you calculate the delta?

Reply:
Yeah it would be susceptible to infinite loops I guess, that could be worked around by keeping a counter though. I'm not sure whether we should be doing it or not, my guess is that not doing it we're rendering a scene that's "half updated" and we may see rendering artifacts.